### PR TITLE
Update release notes and version to correspond to the latest release

### PR DESCRIPTION
**Pull request by vshampor**

### CODE REVIEW

File: ReleaseNotes.md [hunk 1]
```diff
@@ -7,6 +7,33 @@ samples distributed with the code.  The samples demonstrate the usage of compres
...
+
+## New in Release 2.0.1
+Target version updates:
+- Bump target framework versions to PyTorch 1.9.1 and TensorFlow 2.4.3
+- Increased target HuggingFace transformers version for the integration patch to 4.9.1
+
+Bugfixes:
+- Fixed statistic collection for the algo mixing scenario
+- Increased pruning algorithm robustness in cases of a disconnected NNCF graph
+- Fixed the fatality of NNCF graph PNG rendering failures
```
Comment (ljaljushkin) [inline ReleaseNotes.md:22]: whether `fatality`, or `failures` seems to be redundant
```suggestion
- Fixed the fatality of NNCF graph PNG rendering
```

File: ReleaseNotes.md [hunk 2]
```diff
@@ -7,6 +7,33 @@ samples distributed with the code.  The samples demonstrate the usage of compres
...
+- Fixed README command lines
+- (PyTorch) Fixed a bug with quantizing shared weights multiple times
+- (PyTorch) Fixed knowledge distillation failures in CPU-only and DataParallel scenarios
+- (PyTorch) Fixed sparsity application for torch.nn.Embedding and EmbeddingBag modules
+- (PyTorch) Added GroupNorm + ReLU as a fusable pattern
+- (TensorFlow) Fixed gamma fusion handling for pruning TF BatchNorm
+- (PyTorch) Fixed pruning for models where operations have multiple convolution predecessors
+- (PyTorch) Fixed NNCFNetwork wrapper so that `self` in the calls to the wrapped model refer to the wrapper NNCFNetwork object and not to the wrapped model
```
Comment (ljaljushkin) [inline ReleaseNotes.md:30]:
```suggestion
- (PyTorch) Fixed NNCFNetwork wrapper so that `self` in the calls to the wrapped model refers to the wrapper NNCFNetwork object and not to the wrapped model
```

File: ReleaseNotes.md [hunk 3]
```diff
@@ -7,6 +7,33 @@ samples distributed with the code.  The samples demonstrate the usage of compres
...
+Target version updates:
+- Relax TensorFlow version requirements to 2.4.x
+
+## New in Release 2.0.1
+Target version updates:
+- Bump target framework versions to PyTorch 1.9.1 and TensorFlow 2.4.3
+- Increased target HuggingFace transformers version for the integration patch to 4.9.1
+
+Bugfixes:
+- Fixed statistic collection for the algo mixing scenario
```
Comment (ljaljushkin) [inline ReleaseNotes.md:20]: to be consistent with format below it would be great to add (PyTorch, Tensorflow) labels here

File: ReleaseNotes.md [hunk 4]
```diff
@@ -7,6 +7,33 @@ samples distributed with the code.  The samples demonstrate the usage of compres
...
+- (PyTorch) Fixed sparsity application for torch.nn.Embedding and EmbeddingBag modules
+- (PyTorch) Added GroupNorm + ReLU as a fusable pattern
+- (TensorFlow) Fixed gamma fusion handling for pruning TF BatchNorm
+- (PyTorch) Fixed pruning for models where operations have multiple convolution predecessors
+- (PyTorch) Fixed NNCFNetwork wrapper so that `self` in the calls to the wrapped model refer to the wrapper NNCFNetwork object and not to the wrapped model
+- (PyTorch) Fixed tracing of `view` operations to handle shape arguments with the `torch.Tensor` type 
+- (PyTorch) Added matmul ops to be considered for fusing
+- (PyTorch, TensorFlow) Fixed tensorboard logging for accuracy-aware scenarios
+- (PyTorch, TensorFlow) Fixed FLOPS calculation for grouped convolutions
+- (PyTorch) Fixed knowledge distillation failures for tensors of unsupported shapes - will ignore output tensors with unsupported shapes now instead of crashing.
```
Comment (ljaljushkin) [inline ReleaseNotes.md:35]:
```suggestion
- (PyTorch) Fixed knowledge distillation failures for tensors of unsupported shapes - will ignore output tensors with unsupported shapes instead of crashing now.
```

**Review by ljaljushkin — CHANGES_REQUESTED** [review]

File: ReleaseNotes.md (hunk 1 above)
Comment (vshampor) [reply ReleaseNotes.md:22]: I'll have to stand by my own wording here - currently the failures may still occur, it's just that they won't be fatal to the entire script run.

File: ReleaseNotes.md (hunk 2 above)
Comment (vshampor) [reply ReleaseNotes.md:30]: Done

File: ReleaseNotes.md (hunk 3 above)
Comment (vshampor) [reply ReleaseNotes.md:20]: Done

File: ReleaseNotes.md (hunk 4 above)
Comment (vshampor) [reply ReleaseNotes.md:35]: Fixed in a third, more proper way, please review

File: ReleaseNotes.md (hunk 1 above)
Comment (ljaljushkin) [reply ReleaseNotes.md]: ok, got it

**Review by ljaljushkin — APPROVED** [review]
